cmd/drone-server/inject_login.go

```diff
@@ -164,13 +164,21 @@ func provideStashLogin(config config.Config) login.Middleware {
 // provideRefresher is a Wire provider function that returns
 // an oauth token refresher for Bitbucket.
 func provideRefresher(config config.Config) *oauth2.Refresher {
-	if config.Bitbucket.ClientID != "" {
+	switch {
+	case config.Bitbucket.ClientID != "":
 		return &oauth2.Refresher{
 			ClientID:     config.Bitbucket.ClientID,
 			ClientSecret: config.Bitbucket.ClientSecret,
 			Endpoint:     "https://bitbucket.org/site/oauth2/access_token",
 			Source:       oauth2.ContextTokenSource(),
 		}
+	case config.Gitea.Server != "":
```

Comment:
I recommend changing this to `config.Gitea.ClientID != ""` since we only want to enable the refresher if oauth is configured.

Reply:
updated, thanks :)